library/common/engine.cc

```diff
@@ -233,4 +233,16 @@ Http::Client& Engine::httpClient() {
   return *http_client_;
 }
 
+void Engine::flushStats() {
+  // The server will be null if the server has not been initialized as part of main.
+  // In this case, we can simply ignore the flush.
+
+  Thread::LockGuard lock(mutex_);
```

Comment:
These checks/guards are unnecessary once dispatch is moved to main_interface.

Reply:
(You can ASSERT isThreadSafe if you want to be defensive/paranoid though.)

Reply:
Done